bazel-mode.el

```diff
@@ -0,0 +1,240 @@
+(defun sort-strings-longest-first (l)
+  (sort l #'(lambda (s1 s2) (> (length s1) (length s2)))))
+
+;; define several category of keywords
+(setq bazel-keywords
+  '("for"))
+
+(setq android-rules
+  '("android_binary" "aar_import" "android_library" "android_device"
+	"android_ndk_repository" "android_sdk_repository"))
+
+(setq apple-rules
+  '("apple_genrule" "swift_library"))
+
+(setq cc-rules
+  '("cc_binary" "cc_inc_library" "cc_library" "cc_proto_library" "cc_test"))
+
+(setq closure-rules
+  '("closure_css_binary" "closure_css_library" "closure_java_template_library"
+	"closure_js_binary" "closure_js_deps" "closure_js_library"
+	"closure_js_proto_library" "closure_js_template_library" "closure_js_test"
+	"closure_py_template_library" "phantomjs_test"))
+
+(setq d-rules
+  '("d_library" "d_source_library" "d_binary" "d_test" "d_docs"))
+
+(setq docker-rules
+  '("docker_build" "docker_bundle" "docker_pull" "docker_push"))
+
+(setq dotnet-rules
+  '("csharp_binary" "csharp_library" "csharp_nunit_test" "dll_import"
+	"new_nuget_package" "nuget_package"))
+
+(setq extra-actions-rules
+  '("action_listener" "extra_action"))
+
+(setq function-rules
```

Comment:
Please add a reference to https://docs.bazel.build/versions/master/be/functions.html in a comment

Reply:
Done.